internal/monitoring/monitoring.go

```diff
@@ -28,11 +28,49 @@ var (
 		Name:      "blobstore_gc_duration",
 		Help:      "The processing duration a garbage collection in seconds",
 	})
+
+	blobCount = prometheus.NewGauge(prometheus.GaugeOpts{
+		Namespace: "charmstore",
+		Subsystem: "archive",
+		Name:      "blob_count",
+		Help:      "The total number of stored blobs.",
+	})
+
+	maxBlobSize = prometheus.NewGauge(prometheus.GaugeOpts{
+		Namespace: "charmstore",
+		Subsystem: "archive",
+		Name:      "max_blob_size",
+		Help:      "The maximum stored blob size",
+	})
+
+	meanBlobSize = prometheus.NewGauge(prometheus.GaugeOpts{
+		Namespace: "charmstore",
+		Subsystem: "archive",
+		Name:      "mean_blob_size",
+		Help:      "The mean stored blob size",
+	})
 )
 
+type BlobStats struct {
```

Comment:
Docstrings. Also, since this is already called BlobStats, maybe call the fields just Count, MaxSize and MeanSize?